programs/dynamic-bonding-curve/src/state/virtual_pool.rs

```diff
@@ -241,17 +248,59 @@ impl VirtualPool {
         };
 
         let SwapAmount {
+            amount_in: consumed_input_amount, // amount excluded fee
             output_amount,
             next_sqrt_price,
         } = match trade_direction {
             TradeDirection::BaseToQuote => {
                 self.get_swap_amount_from_base_to_quote(config, actual_amount_in)
             }
             TradeDirection::QuoteToBase => {
-                self.get_swap_amount_from_quote_to_base(config, actual_amount_in)
+                self.get_swap_amount_from_quote_to_base(config, actual_amount_in, swap_mode)
             }
         }?;
 
+        // check if it is partial fill
+        let user_pay_input_amount = if consumed_input_amount < actual_amount_in {
+            if fee_mode.fees_on_input {
+                FeeRateLimiter::revert_if_limiter_applied(
+                    &config.pool_fees.base_fee,
+                    trade_fee_numerator,
+                )?;
+
+                let amount_in_include_fee = PoolFeesConfig::get_included_fee_amount(
```

Comment:
I prefer the name `included_fee_amount_in` more

Reply:
Fixed